dev_apps/in_app_pay_test/js/app.js

```diff
@@ -0,0 +1,166 @@
+      url: '/jwt-source?server=' + server + '&_=' + Date.now()
+    });
+  }
+
+  var App = {
+    run: function() {
+      ['jwt-input',
+       'server',
+       'error',
+       'jwt',
+       'refresh',
+       'log',
+       'log-content',
+       'jwt-panel',
+       'jwt-content',
+       'pay',
+       'toggle'].forEach(id => {
+        this[toCamelCase(id)] = document.getElementById(id);
+      });
+
+      refresh.addEventListener('click', App.refresh);
+      pay.addEventListener('click', App.pay);
+      toggle.addEventListener('click', () => {
+        if (jwtPanel.classList.contains('hidden')) {
+          App.showJWT();
+        } else {
+          App.showLog();
+        }
+      });
+      server.addEventListener('change', App.refresh);
+      App.refresh();
+    },
+
+    pay: function() {
+      App.clearLog();
+      App.showLog();
+      App.writeLog('Signing JWT...');
+      fetch({
+        method: 'POST',
+        url: '/pay',
+        body: {
+          jwt: jwtContent.textContent,
+          server: server.value
+        }
+      }).then((response) => {
+        App.writeLog('[Transaction ID] -> ' + response.transID);
+        App.writeLog('Calling navigator.mozPay...');
+        return new Promise((resolve, reject) => {
+          var req = navigator.mozPay([response.jwt]);
+          req.onsuccess = () => {
+            resolve(response.transID);
+          };
+          req.onerror = () => {
+            reject('Payment canceled ' + req.error.name);
+          };
+        });
+      }).then((transactionID) => {
+        App.writeLog('Payment done!');
+        App.verifyTransaction(transactionID, 0);
+      }).catch((error) => {
+        App.writeLog('Oh crap ' + error);
+      });
+    },
+
+    verifyTransaction(transactionID, retries) {
+      App.writeLog('Verifying transaction... Retry ' + retries);
+      fetch({
+        method: 'GET',
+        url: '/transaction/' + transactionID
+      }).then((response) => {
+        if (response.state == 'completed') {
+          App.writeLog('Payment verified!');
+          return;
+        }
+        if (retries > MAX_RETRIES) {
+          App.writeLog('Payment could not be verified :(');
+          return;
+        }
+        setTimeout(() => {
+          App.verifyTransaction(transactionID, retries++);
+        }, 1000);
+      });
+    },
+
+    showLog: function() {
+      log.classList.remove('hidden');
+      jwtPanel.classList.add('hidden');
+      toggle.textContent = 'Show JWT';
+    },
+
+    showJWT: function() {
+      jwtPanel.classList.remove('hidden');
+      log.classList.add('hidden');
+      toggle.textContent = 'Show Log';
+    },
+
+    writeLog: function(msg) {
+      console.log(msg);
+      logContent.textContent += '\n' + msg;
+    },
+
+    clearLog: function() {
+      logContent.textContent = '';
+    },
+
+    refresh: function() {
+      getJWT(server.value).then((jwt) => {
+        jwtContent.textContent = JSON.stringify(jwt);
```

Comment:
ok, I think I see. The code would read better to me if you attached the doc references to a top-level var like ``pageDocs.jwtContent.textContent = ...``